worker/pkg/workflows/datasync/activities/activities.go

```diff
@@ -175,9 +175,15 @@ func (b *benthosBuilder) buildBenthosSqlSourceConfigResponses(
 			return nil, err
 		}
 
-		if (processorConfig.Mutation != nil && *processorConfig.Mutation != "") ||
-			(processorConfig.Javascript != nil && processorConfig.Javascript.Code != "") {
-			bc.StreamConfig.Pipeline.Processors = append(bc.StreamConfig.Pipeline.Processors, *processorConfig)
+		for _, pc := range processorConfig {
+			mutationValid := pc.Mutation != nil && *pc.Mutation != ""
+
+			javascriptValid := pc.Javascript != nil && pc.Javascript.Code != ""
+
+			if mutationValid || javascriptValid {
+				bc.StreamConfig.Pipeline.Processors = append(bc.StreamConfig.Pipeline.Processors, *pc)
+			}
+
 		}
```

Comment:
this might be a little overkill. it's probably save to trust the output of `buildProcessorConfig` to ensure that the contents are correct.
This can then just become 

```
processorConfig, err := b.buildProcessorConfig(ctx, tableMappings.Mappings)
if err != nil {
  return err
}
bc.StreamConfig.Pipeline.Processors = append(bc.StreamConfig.Pipeline.Processors, *processorConfig)
```

Reply:
Removed extra checks. Since processorConfigs returns a slice so we have to range over it and append each object to the processors.